plugins/cachekey/pattern.h

```diff
@@ -18,29 +18,23 @@
 
 /**
  * @file pattern.h
- * @brief PRCE related classes (header file).
+ * @brief Regex related classes (header file).
  */
 
 #pragma once
 
 #include "tscore/ink_defs.h"
-
-#ifdef HAVE_PCRE_PCRE_H
-#include <pcre/pcre.h>
-#else
-#include <pcre.h>
-#endif
+#include "tsutil/Regex.h"
 
 #include "common.h"
 
 /**
- * @brief PCRE matching, capturing and replacing
+ * @brief Regex matching, capturing and replacing
  */
 class Pattern
 {
 public:
-  static const int TOKENCOUNT = 10;             /**< @brief Capturing groups $0..$9 */
-  static const int OVECOUNT   = TOKENCOUNT * 3; /**< @brief pcre_exec() array count, handle 10 capture groups */
+  static const int TOKENCOUNT = 10; /**< @brief Capturing groups $0..$9 */
 
   Pattern();
   virtual ~Pattern();
```

Comment:
I don't think there's inheritance of Pattern, right? We should be able to remove `virtual` and set this ` = default;` here. And remove `Pattern::~Pattern() {}` in the .cc file.